### ISSUE TITLE
Adding level crossing from validation causes overlaping railway

### ISSUE THREAD

**Issue by paulklie** [openstreetmap/iD#9224]
### URL

https://www.openstreetmap.org/way/205764563

### How to reproduce the issue?

Here you can see a railway crossing a road without a crossing node.
If you open this rail with iD it flags this error.
The autoresolve "connect the features" however merges the two crossings, and thus causes the left and right branch of the railway to overlap.
Worst of all: This then is not flagged as an error, and the user cannot see the overlap.

### Screenshot(s) or anything else?

![image](https://user-images.githubusercontent.com/73854742/181786533-86f47520-819e-463b-a954-613fb8cf4821.png)
![image](https://user-images.githubusercontent.com/73854742/181786596-6938c9a0-c543-4f09-a512-c353d8a26b5e.png)

### Which deployed environments do you see the issue in?

Released version at openstreetmap.org/edit

### What version numbers does this issue effect?

2.21.1

**Comment by paulklie** [comment]
A potentially even worse situation is when two (very close) parallel rail lines are merged in the same way:
![image](https://user-images.githubusercontent.com/73854742/181788432-93a67123-dc7b-4297-bd4a-a4cadd5c7b79.png)
![image](https://user-images.githubusercontent.com/73854742/181788476-e3b939d3-2950-4c8f-8c17-38d8d89d0dbe.png)
I think the best way to avoid this is to not merge railway crossing at all.

**Comment by tyrasd** [comment]
Confirming. And this can also happen for other types of intersections, e.g. road-footway. At first glance, I don't see what the intention behind the merging of closeby intersection nodes is, but maybe there are some use cases that benefit from it. However, it should not lead to topological errors such as overlapping way segments.

**Comment by 1ec5** [comment]
The original example has since been [resolved](https://www.openstreetmap.org/changeset/124249765), moving the crossing nodes to be about half a meter apart.

The validation fix attempts to reuse a node within 0.75&nbsp;meters (2½&nbsp;ft.):

https://github.com/openstreetmap/iD/blob/3079767363bff73da197984d0a4835bb5bfc65f8/modules/validations/crossing_ways.js#L754 https://github.com/openstreetmap/iD/blob/3079767363bff73da197984d0a4835bb5bfc65f8/modules/validations/crossing_ways.js#L759-L760

This isn’t very far, though it’s a little farther than the 20-centimeter threshold for the close-nodes validator warning:

https://github.com/openstreetmap/iD/blob/3079767363bff73da197984d0a4835bb5bfc65f8/modules/validations/close_nodes.js#L12

Originally, any node would get repurposed, but #8625 narrowed that behavior to just crossing nodes. This was intentional, because often a pedestrian crossing is mapped first as a node, then as a way later on by another mapper. We wouldn’t want the crossing ways fix to produce a redundant crossing node that the latter mapper might not even see due to overlapping icons. Level crossings are less likely to be mapped in two stages like this, so maybe the heuristic could be tightened to only affect pedestrian crossings.

**Comment by tyrasd** [comment]
> This was intentional, because often a pedestrian crossing is mapped first as a node, then as a way later on by another mapper.

Yes, that's what I thought as well. The fix should however not change the topology of other ways (nor lead to topological errors like overlapping ways) IMO.
